patches/minecraft/net/minecraft/entity/item/BoatEntity.java.patch

```diff
@@ -30,3 +69,61 @@
     public static enum Status {
        IN_WATER,
        UNDER_WATER,
+@@ -786,22 +782,49 @@
+       IN_AIR;
+    }
+ 
+-   public static enum Type {
+-      OAK(Blocks.field_196662_n, "oak"),
+-      SPRUCE(Blocks.field_196664_o, "spruce"),
+-      BIRCH(Blocks.field_196666_p, "birch"),
+-      JUNGLE(Blocks.field_196668_q, "jungle"),
+-      ACACIA(Blocks.field_196670_r, "acacia"),
+-      DARK_OAK(Blocks.field_196672_s, "dark_oak");
++   public static enum Type implements net.minecraftforge.common.IExtensibleEnum {
++      OAK(Blocks.field_196662_n, "oak", () -> Items.field_151124_az),
++      SPRUCE(Blocks.field_196664_o, "spruce", () -> Items.field_185150_aH),
++      BIRCH(Blocks.field_196666_p, "birch", () -> Items.field_185151_aI),
++      JUNGLE(Blocks.field_196668_q, "jungle", () -> Items.field_185152_aJ),
++      ACACIA(Blocks.field_196670_r, "acacia", () -> Items.field_185153_aK),
++      DARK_OAK(Blocks.field_196672_s, "dark_oak", () -> Items.field_185154_aL);
+ 
+       private final String field_184990_g;
+       private final Block field_195934_h;
++      private final java.util.function.Supplier<net.minecraft.item.ItemStack[]> dropSupplier;
++      private java.util.function.Supplier<Item> boatItemSupplier;
+ 
+-      private Type(Block p_i48146_3_, String p_i48146_4_) {
++      private Type(Block p_i48146_3_, String p_i48146_4_, java.util.function.Supplier<Item> boatItemSupplier) {
+          this.field_184990_g = p_i48146_4_;
+          this.field_195934_h = p_i48146_3_;
++         this.boatItemSupplier = boatItemSupplier;
++         this.dropSupplier = () -> new net.minecraft.item.ItemStack[]{ new net.minecraft.item.ItemStack(Items.field_151055_y, 2), new net.minecraft.item.ItemStack(p_i48146_3_, 3) };
+       }
+ 
++      private Type(Block planks, String name, java.util.function.Supplier<Item> boatItemSupplier, java.util.function.Supplier<net.minecraft.item.ItemStack[]> drops) {
++         this.field_184990_g = name;
++         this.field_195934_h = planks;
++         this.boatItemSupplier = boatItemSupplier;
++         this.dropSupplier = drops;
++      }
++
++      public static Type create(String enumName, Block planks, String name, java.util.function.Supplier<Item> boatItemSupplier) {
++         throw new IllegalStateException("Enum not extended");
++      }
++
++      public static Type create(String enumName, Block planks, String name, java.util.function.Supplier<Item> boatItemSupplier, java.util.function.Supplier<net.minecraft.item.ItemStack[]> drops) {
++         throw new IllegalStateException("Enum not extended");
++      }
++
++      public Item getBoatItem() {
++         return this.boatItemSupplier.get();
++      }
++
++      public com.google.common.collect.ImmutableSet<net.minecraft.item.ItemStack> getDrops() {
```

Comment:
Typically we use Lists not sets to not limit duplication, this also doesn't have the same behavior of vanilla. Namely it'll toss out two entities that may or may not merge. Instead of 5 entities in random directions.